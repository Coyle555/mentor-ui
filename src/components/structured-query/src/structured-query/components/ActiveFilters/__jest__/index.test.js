import React from 'react';
import { ActiveFiltersComponent } from '../index';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render, wait } from '@testing-library/react';

afterEach(cleanup);

describe('Rendering active filters', () => {
	test('No search tokens', () => {
		const tree = renderer.create(<ActiveFiltersComponent />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Some search tokens passed in', () => {
		const tree = renderer.create(
			<ActiveFiltersComponent searchTokens={[{ id: 'foo', label: 'Foo' }]} />
		).toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Opening active filters with search tokens', async () => {
		const { container, queryByText } = render(
			<ActiveFiltersComponent searchTokens={[{
				id: 'foo',
				label: 'Foo',
				value: 'value'
			}]} />);

		fireEvent.click(container.querySelector('span.left-addon.active-filter-container'));

		await wait(() => {
			expect(queryByText('value')).toBeTruthy();
			expect(queryByText('Clear All')).toBeTruthy();
		});
	});
});

describe('Handle click outside -- 3rd party lib', () => {

	test('Handle click outside method', () => {
		const instance = renderer.create(<ActiveFiltersComponent />).getInstance();

		instance.state.filtersActive = true;
		instance.handleClickOutside();
		expect(instance.state.filtersActive).toBe(false);
	});
});
