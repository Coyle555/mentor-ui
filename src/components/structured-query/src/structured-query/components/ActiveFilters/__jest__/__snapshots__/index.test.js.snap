// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering active filters No search tokens 1`] = `
<span
  className="left-addon active-filter-container"
  data-for="structured-query-tooltip"
  data-tip="View Filters"
  onClick={[Function]}
>
  <i
    className="far fa-list"
  />
  <span
    className="active-filter-count"
  >
    0
  </span>
</span>
`;
